Review

PietrH opened the pull request:
# Ready

PietrH committed:
Do not include raw data for now

PietrH commented:
In the source spreadsheet, on the Herb tab, column J, row 1898, there is a text entry where we expect a number for `Cover`, resulting in an NA in the mapping. 

``` r
directory <- file.path("datasets", "fornalab-formica-wp1-vegetation", "data")
readxl::read_excel(
here::here(directory, "raw", "Vegetation survey_version37.xlsx"),
  sheet = "Herb",
  col_types = "text"
) |> dplyr::slice(1897:1899) |> dplyr::glimpse()
#> Rows: 3
#> Columns: 15
#> $ Code              <chr> "NOMET2P1", "NOMET2P1", "NOMET2P2"
#> $ Latitude          <chr> "62.5635474", "62.5635474", "62.5635292"
#> $ Longitude         <chr> "9.0788985", "9.0788985", "9.0788217"
#> $ Date              <chr> "43271.0", "43271.0", "43271.0"
#> $ Region            <chr> "NO", "NO", "NO"
#> $ Elevation         <chr> "ME", "ME", "ME"
#> $ Type              <chr> "T2", "T2", "T2"
#> $ Plot              <chr> "P1", "P1", "P2"
#> $ Species           <chr> "Ast.spe", "Ger.syl", "Sta.syl"
#> $ Cover             <chr> "0.5", "r", "12.0"
#> $ Flowering         <chr> NA, NA, NA
#> $ Fruiting          <chr> NA, NA, NA
#> $ `Flow-fru`        <chr> NA, NA, NA
#> $ Species_name_long <chr> "Asteraceae spec.", "Geranium sylvaticum", "Stachys …
#> $ Nr                <chr> "1900.0", "1901.0", "1902.0"
```

<sup>Created on 2024-12-16 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

PietrH committed:
avoid mixed dplyr and base syntax for wrangling

PietrH committed:
remove hanging comma

PietrH committed:
Taken from https://dwc.tdwg.org/terms/#dwc:sampleSizeUnit

PietrH commented on datasets/fornalab-formica-wp1-vegetation/src/dwc_mapping.Rmd:
```diff
@@ -175,7 +179,7 @@ event <-
     eventDate = .data$Date_of_survey,
     samplesizeValue = 9,
-    sampleSizeUnit = "m2",
+    sampleSizeUnit = "square metre",
```
I had a look here https://dwc.tdwg.org/terms/#dwc:sampleSizeUnit and it's written out, I wonder what's actually in use on GBIF though. It'd be interesting to see what values are actually in use.

PietrH commented on datasets/fornalab-formica-wp1-vegetation/src/dwc_mapping.Rmd:
```diff
@@ -143,10 +143,14 @@ data_tree <-
   )
 
 column_names <- c("occurrence_id", "Cover", "organism_quantity_type", "event_id", "Species_name_long")
-data_veg <- rbind(
```
See chat: try to avoid mixing base and dplyr syntax when wrangling data. It's harder to read that way. 

Here it's more defensive to use tidyselect and dplyr select so we'll get an error when a column is missing from a table.

PietrH commented on .gitignore:
```diff
@@ -48,6 +48,7 @@ docs/
 *.txt
 *.gz
 occurrence.RData
+/datasets/fornalab-formica-wp1-vegetation/data/raw
```
Avoid pushing any raw data to Github, there are some unneccesairy lines in .gitignore

PietrH commented on datasets/fornalab-formica-wp1-vegetation/src/dwc_mapping.Rmd:
```diff
@@ -175,7 +179,7 @@ event <-
+    sampleSizeUnit = "square metre",
     habitat = "temperate forest",
     samplingProtocol = "vegetation survey",
     country = dplyr::recode(
```
If a country is missing/unmapped, you will not throw an error here. 

There is a package to do exactly this; `countrycode`, it's very convenient if you ever need to map many countrycodes.

sannegovaert replied:
These codes are specific to this project. We had 2 regions in Sweden en used SS (south Sweden) en CS (central Sweden).
But good to know!

PietrH commented:
Do you have a `identificationVerificationStatus`? It's not a very commonly used term, but if you know the identifications were verified, then it's nice to include it.

sannegovaert commented:
> In the source spreadsheet, on the Herb tab, column J, row 1898, there is a text entry where we expect a number for `Cover`, resulting in an NA in the mapping.
> 
> ```r
> directory <- file.path("datasets", "fornalab-formica-wp1-vegetation", "data")
> readxl::read_excel(
> here::here(directory, "raw", "Vegetation survey_version37.xlsx"),
>   sheet = "Herb",
>   col_types = "text"
> ) |> dplyr::slice(1897:1899) |> dplyr::glimpse()
> #> Rows: 3
> #> Columns: 15
> #> $ Code              <chr> "NOMET2P1", "NOMET2P1", "NOMET2P2"
> #> $ Latitude          <chr> "62.5635474", "62.5635474", "62.5635292"
> #> $ Longitude         <chr> "9.0788985", "9.0788985", "9.0788217"
> #> $ Date              <chr> "43271.0", "43271.0", "43271.0"
> #> $ Region            <chr> "NO", "NO", "NO"
> #> $ Elevation         <chr> "ME", "ME", "ME"
> #> $ Type              <chr> "T2", "T2", "T2"
> #> $ Plot              <chr> "P1", "P1", "P2"
> #> $ Species           <chr> "Ast.spe", "Ger.syl", "Sta.syl"
> #> $ Cover             <chr> "0.5", "r", "12.0"
> #> $ Flowering         <chr> NA, NA, NA
> #> $ Fruiting          <chr> NA, NA, NA
> #> $ `Flow-fru`        <chr> NA, NA, NA
> #> $ Species_name_long <chr> "Asteraceae spec.", "Geranium sylvaticum", "Stachys …
> #> $ Nr                <chr> "1900.0", "1901.0", "1902.0"
> ```
> 
> Created on 2024-12-16 with [reprex v2.1.0](https://reprex.tidyverse.org)

Well seen! I think this means <0.5% but I'll check it and solve this.

sannegovaert commented:
> identificationVerificationStatus

Didn't know about this term.
The [provided example](https://dwc.tdwg.org/list/#dwc_identificationVerificationStatus) is "0" (unverified). 
So then "1" indicates verified?

PietrH committed:
NA to emtpy string

PietrH commented:
releve: for only one event the total coverage is the same as the sums of the coverages: 

```r
readr::read_csv("datasets/fornalab-formica-wp1-vegetation/data/processed//releve.csv") |>
  # dplyr::rowwise() |>
  # dplyr::mutate(Total_calculated = sum(
  #                 dplyr::c_across(
  #                   dplyr::matches("(?<!Total)InPercentage$", perl = TRUE)))) |>
  # dplyr::select(dplyr::contains("Total"))
  dplyr::rowwise() |>
  dplyr::mutate(
    Total_calculated =
      sum(
        coverTreesInPercentage,
        coverShrubsInPercentage,
        coverHerbsInPercentage,
        coverMossesInPercentage,
        coverLitterInPercentage,
        na.rm = TRUE
      )
  ) |>
  dplyr::select(
    eventID,
    Total_calculated,
    coverTotalInPercentage
  ) |> 
  dplyr::filter(Total_calculated != coverTotalInPercentage)
```

PietrH committed:
remove `legalcode` suffix from license

PietrH commented:
Is there a ror.org `institutionID` for Ghent University?

PietrH commented:
I removed the `/legalcode` suffix from the `license` fields, this is not obligatory and GBIF will add it back in. See team chat. 

```r

remove_legalcode_from_license <- function(file_path) {
  readr::read_csv(file_path,
                  col_types = readr::cols(.default = readr::col_character())) |>
    dplyr::mutate(
      license = stringr::str_remove(license, stringr::fixed("legalcode"))
    ) |>
    readr::write_csv(file_path, na = "")
  
}

remove_legalcode_from_license("datasets/fornalab-formica-wp1-vegetation/data/processed/event.csv")
remove_legalcode_from_license("datasets/fornalab-formica-wp1-vegetation/data/processed/occurrence.csv")

```

PietrH commented:
The occurrenceID is very long, and encodes a lot of information. I prefer shorter id's, and not encoding any information at all (so they are easier for humans to read and type, and so no id/data mismatches can occur if the data ever changes). 

In this case neither is really a problem. But something to keep in mind.

PietrH commented:
~~Should we add `bibliographicCitation` or `associatedReferences`? These make it easier for people to find relevant info if they come across an occurrence in isolation (via gbif api request for example).~~

PietrH commented:
~~Small suggestion; add a `continent`, this will avoid the `Continent derived from coordinates` and `Continent derived from country` issues on gbif (which are more like notices than issues really)~~

Edit: countryCode is way better idea, thanks Peter

PietrH commented:
There are some issues with scientificName matching: 

```r
# Trouble in scientificNames
from_gbif <-
  occurrence %>%
  select(scientificName, taxonRank) %>%
  distinct %>%
  mutate(
    lookup = purrr::map2(
      scientificName,
      taxonRank,
      ~ rgbif::name_backbone(.x, rank = .y, kingdom = "Plantae")
    ),
    .keep = "none"
  ) %>%
  dplyr::pull(lookup) %>%
  purrr::list_rbind()

## not matched exactly
unique(occurrence$scientificName)[!unique(occurrence$scientificName) %in% from_gbif$canonicalName]
```

You ought to look into this, there might be some errors in your scientific names. 

From the dwc archive validator on gbif.org:

![image](https://github.com/user-attachments/assets/e0e838ff-5f81-4ca9-a70b-5ad811c95351)


Problematic scientificNames that jumped out to me:

- "#N/A"
- spec. is not a valid scientificName
- numbers in scientific names
- Entering plant hybrids is tricky, I always need to look up how to do it.

PietrH commented:
You seem to be encoding absence records with the scientific name: "No species present" , I don't think this is intentional

PietrH commented:
@sannegovaert Looks pretty good! The only critical thing is the taxon name matching. Let me know if you need help.

sannegovaert committed:
Changes from commits from Pieter

- remove /legalcode in license url
- set sampleSizeUnit to 'square meters'
- remove "NA" in text

sannegovaert committed:
correction species names

sannegovaert committed:
match species names with GBIF backbone

sannegovaert committed:
add data quality check

sannegovaert committed:
remove "NA" in raw data

sannegovaert committed:
correct species names to match with GBIF backbone

sannegovaert committed:
add project metadata

sannegovaert committed:
IPT does not support the html tag <i> anymore

sannegovaert committed:
add missing coordinates in raw data 'plot characteristics' + make coverPercentage all numeric

sannegovaert committed:
html tag <i> is not accepted by IPT anymore

sannegovaert committed:
add occurrene_id to raw data + small update abstract

sannegovaert reviewed (approved):
Great, thank you for your help Pieter!
